src/borg/_hashindex.c

```diff
@@ -481,33 +632,34 @@ hashindex_free(HashIndex *index)
     free(index);
 }
 
-#ifndef BORG_NO_PYTHON
-static void
-hashindex_write(HashIndex *index, PyObject *file_py)
+int
+write_hashheader(HashIndex *index, PyObject *file_py)
 {
-    PyObject *length_object, *buckets_view, *tmp;
+    PyObject *length_object, *tmp;
     Py_ssize_t length;
-    Py_ssize_t buckets_length = (Py_ssize_t)index->num_buckets * index->bucket_size;
+
     HashHeader header = {
         .magic = MAGIC,
+        .version = _htole32(2),
         .num_entries = _htole32(index->num_entries),
         .num_buckets = _htole32(index->num_buckets),
-        .key_size = index->key_size,
-        .value_size = index->value_size
+        .num_empty = _htole32(index->num_empty),
+        .key_size = _htole32(index->key_size),
+        .value_size = _htole32(index->value_size)
     };
```

Comment:
My C is a bit rusty, does this zero-initialize the `reserved` space? Would be good to have it always be zero to allow for extensions.

Reply:
I guess the extension mechanism would be to increase the version number.

Then we don't have to care about it being 0, right?

Reply:
Explicitly zeroing it avoids writing random stack contents to disk still, which is probably a good idea.

Reply:
yeah, indeed.

Reply:
not sure what we have as assured default behaviour though.

Reply:
done: init reserved to zero before writing to disk.